### PR TITLE
WRN-1372: Refactored Item unit tests based on testing-library best practices

### PR DESCRIPTION
**Pull request by daniel-stoian-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Refactored Item unit tests based on testing-library best practices

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRN-1372

### Comments

Enact-DCO-1.0-Signed-off-by: Daniel Stoian daniel.stoian@lgepartner.com

### CODE REVIEW

File: Item/tests/Item-specs.js [hunk 1]
```diff
@@ -9,96 +9,97 @@ describe('Item Specs', () => {
...
 
 		expect(actual).toBeInTheDocument();
 	});
 
 	test('should support `slotBefore`', () => {
 		const expected = 'slot before';
 
-		const {getByText} = render(<ItemBase data-testid="item" slotBefore={expected}>Hello Item</ItemBase>);
-		const actual = getByText(expected);
+		render(<ItemBase data-testid="item" slotBefore={expected}>Hello Item</ItemBase>);
```
Comment (stanca-pop-lgp) [inline Item/tests/Item-specs.js:48]: Do you still need data-testid?

File: Item/tests/Item-specs.js [hunk 2]
```diff
@@ -9,96 +9,97 @@ describe('Item Specs', () => {
...
 
 		expect(actual).toBeInTheDocument();
 	});
 
 	test('should support `slotAfter`', () => {
 		const expected = 'slot after';
 
-		const {getByText} = render(<ItemBase data-testid="item" slotAfter={expected}>Hello Item</ItemBase>);
-		const actual = getByText(expected);
+		render(<ItemBase data-testid="item" slotAfter={expected}>Hello Item</ItemBase>);
```
Comment (stanca-pop-lgp) [inline Item/tests/Item-specs.js:57]: Here as well.

File: Item/tests/Item-specs.js [hunk 3]
```diff
@@ -9,96 +9,97 @@ describe('Item Specs', () => {
...
+		expect(actual).not.toHaveClass(expected);
 	});
 
 	test('should add the selected class when given the selected prop', () => {
-		const {getByTestId} = render(<ItemBase data-testid="item" selected>Hello Item</ItemBase>);
+		render(<ItemBase data-testid="item" selected>Hello Item</ItemBase>);
 
 		const expected = css.selected;
-		const actual = getByTestId('item').className;
+		const actual = screen.getByTestId('item');
```
Comment (stanca-pop-lgp) [inline Item/tests/Item-specs.js:83]: Could you use getByText instead of getByTestId?

Comment (daniel-stoian-lgp) [reply Item/tests/Item-specs.js:83]: In this case no. with getByText a child element is returned and the className 'selected' is assigned to the parent container of the component

**Review by stanca-pop-lgp — APPROVED** [review]
LGTM

**Pull request merged**